applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -583,6 +587,69 @@ class EigensolverStrategy
     ///@name Private Operations
     ///@{
 
+    /**
+     * @brief Recover the solution related to the slave dofs in case of master-slave constraints.
+     * @details This function is an adaptation of the implementation in ConstraintUtilities,
+     *  since there the variables are assumed to be stored in SolutionStepValue.
+     *  Beware that this implementation is only valid for Block B&S, since the master-slave constraints
+     *  don't work with Elimination B&S yet.
+     */
+    void ReconstructSolution(
+        DenseMatrixType& rEigenvectors
+    )
+    {
+        KRATOS_TRY
+
+        const auto& rModelPart = BaseType::GetModelPart();
+        const auto& r_master_slave_constraints = rModelPart.MasterSlaveConstraints();
+        const std::size_t number_of_constraints = rModelPart.NumberOfMasterSlaveConstraints();
+        const std::size_t number_of_eigenvalues = rEigenvectors.size1();
+
+        for (std::size_t i_eigenvalue = 0; i_eigenvalue < number_of_eigenvalues; ++i_eigenvalue){
+
+            // Reset slave dofs
+            for (std::size_t i_const = 0; i_const < number_of_constraints; ++i_const) {
+                auto it_const = r_master_slave_constraints.begin() + i_const;
+                const auto& r_slave_dofs_vector = it_const->GetSlaveDofsVector();
+                for (IndexType i = 0; i < r_slave_dofs_vector.size(); ++i)
+                    rEigenvectors(i_eigenvalue, r_slave_dofs_vector[i]->EquationId()) = 0.0;
+            }
```

Comment:
can you use the parallel utilities please?

Reply:
done